verilog/parser/verilog_token.cc

```diff
@@ -0,0 +1,58 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "verilog/parser/verilog_token.h"
+
+#include <string>
+
+#include "verilog/parser/verilog_parser.h"
+#include "absl/strings/string_view.h"
+
+namespace verilog {
+
+// Strings returned by this function are used as token tags
+// in verible-verilog-syntax' JSON output. Changing them might
+// break third-party code.
```

Comment:
@hzeller We are now exposing the enum names of tokens from `verilog.y` publicly, and this might limit our freedom to change/update these as we see fit.  Or we need to alert users of their fragility.  How should we communicate this?

Reply:
Unless we get rid of TokenTypeToString, it is possible to add overrides for backward compatibility here.

Also, in the current version of the function, only tokens without alias and those explicitly overriden in `verilog_token.cc` use enum name, so most names will not be locked. Almost all used aliases are just Verilog's symbols (e.g. "module", "++", etc), so I guess they'll never change. However, some tokens for which enum name is used are very likely to be used much in 3rd party code (e.g. "SymbolIdentifier").

Reply:
I do think we should mention the potential problem in the documentation, that the structure is not fixed.
But I also think that this is probably less of an issue in real life as

  (a) verilog.y is pretty stable at this point and no drastic long-term changes are to be expected
  (b) people using json output probably use it in some ephemeral context ('playing around with things')